website/src/_layouts/_base.njk

```diff
@@ -19,14 +19,26 @@
 	<meta name="twitter:card" content="summary_large_image" />
 	<meta name="twitter:site" content="@phpstan" />
 	<meta name="twitter:creator" content="@ondrejmirtes" />
-	<meta name="twitter:title" content="{{ title }}" />
+	<meta name="twitter:title" content="{{ title }}"/>
 	<script src="https://cdn.usefathom.com/script.js" data-site="UDESFNRN" defer></script>
+
 	<script>
-		document.documentElement.classList.toggle(
-			'dark',
-			localStorage.getItem('color-theme') === 'dark' || (!('color-theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)
-		);
+		window.ThemeSwitcher = {
+			storageKey: 'color-theme',
+			reloadTheme: () => {
+				let theme = localStorage.getItem(ThemeSwitcher.storageKey);
+				if (theme === 'auto') {
+					theme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+				}
+				document.documentElement.classList.toggle('dark', theme === 'dark');
+				// TODO This is only useful to be caught by other components to do the switch.
```

Comment:
When I get some time, I can also look into implementing the codemirror changes too.